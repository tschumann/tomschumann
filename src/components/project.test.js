import React from 'react';
import renderer from 'react-test-renderer';
import Project from './project';

test('test component renders with language', () => {
	const component = renderer.create(<Project name="Name" body="Body" sourceLink="https://dev.lan" language="cpp" key="1"/>);
	const tree = component.toJSON();

	expect(tree).toMatchSnapshot();
});

test('test component renders without language', () => {
	const component = renderer.create(<Project name="Name" body="Body" sourceLink="https://dev.lan" language="" key="1"/>);
	const tree = component.toJSON();

	expect(tree).toMatchSnapshot();
});

