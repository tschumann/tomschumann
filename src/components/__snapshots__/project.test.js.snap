// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test component renders with language 1`] = `
<div>
  <h3>
    Name
  </h3>
  <div
    className="project-cpp project-description"
    dangerouslySetInnerHTML={
      Object {
        "__html": "Body",
      }
    }
  />
  <div>
    Source code: 
    <a
      href="https://dev.lan"
      target="_blank"
    >
      https://dev.lan
    </a>
  </div>
</div>
`;

exports[`test component renders without language 1`] = `
<div>
  <h3>
    Name
  </h3>
  <div
    className=" project-description"
    dangerouslySetInnerHTML={
      Object {
        "__html": "Body",
      }
    }
  />
  <div>
    Source code: 
    <a
      href="https://dev.lan"
      target="_blank"
    >
      https://dev.lan
    </a>
  </div>
</div>
`;
